crates/rust-analyzer/src/bin/logger.rs

```diff
@@ -63,93 +57,28 @@ impl Logger {
             })
             .ok();
 
-        let chalk_layer = HierarchicalLayer::default()
-            .with_indent_lines(true)
-            .with_ansi(false)
-            .with_indent_amount(2)
-            .with_writer(io::stderr);
-
         let writer = match self.file {
             Some(file) => BoxMakeWriter::new(Arc::new(file)),
             None => BoxMakeWriter::new(io::stderr),
         };
-        let ra_fmt_layer =
-            tracing_subscriber::fmt::layer().event_format(LoggerFormatter).with_writer(writer);
-
-        match chalk_level_dir {
-            Some(val) => {
-                Registry::default()
-                    .with(
-                        self.filter
-                            .add_directive(format!("chalk_solve={}", val).parse()?)
-                            .add_directive(format!("chalk_ir={}", val).parse()?)
-                            .add_directive(format!("chalk_recursive={}", val).parse()?),
-                    )
-                    .with(ra_fmt_layer)
-                    .with(chalk_layer)
-                    .init();
-            }
-            None => {
-                Registry::default().with(self.filter).with(ra_fmt_layer).init();
-            }
-        };
-
-        Ok(())
-    }
-}
-
-#[derive(Debug)]
-struct LoggerFormatter;
-
-impl<S, N> FormatEvent<S, N> for LoggerFormatter
-where
-    S: Subscriber + for<'a> LookupSpan<'a>,
-    N: for<'a> FormatFields<'a> + 'static,
-{
-    fn format_event(
-        &self,
-        ctx: &FmtContext<'_, S, N>,
-        mut writer: Writer,
-        event: &Event<'_>,
-    ) -> fmt::Result {
-        // Write level and target
-        let level = *event.metadata().level();
-
-        // If this event is issued from `log` crate, then the value of target is
-        // always "log". `tracing-log` has hard coded it for some reason, so we
-        // need to extract it using `normalized_metadata` method which is part of
-        // `tracing_log::NormalizeEvent`.
-        let target = match event.normalized_metadata() {
```

Comment:
This comment seems to be either wrong or outdated from my testing so we don't need this custom formatter any longer

Reply:
Actually nevermind, it still has this behaviour? Hmm, that is a bit annoying. Guess we need some changes to land in tracing-tree before we can properly use it for our things.
https://github.com/tokio-rs/tracing/blob/master/tracing-log/src/lib.rs#L165-L193
https://github.com/tokio-rs/tracing/blob/master/tracing-log/src/lib.rs#L277-L320

Reply:
Looking back into this the reason is described here https://docs.rs/tracing-log/latest/tracing_log/trait.NormalizeEvent.html